server/sonar-db-dao/src/main/java/org/sonar/db/component/BranchDao.java

```diff
@@ -74,6 +74,10 @@ public int updateMainBranchName(DbSession dbSession, String projectUuid, String
     return Optional.ofNullable(mapper(session).selectByUuid(uuid));
   }
 
+  public boolean existsNonMainBranch(DbSession dbSession) {
```

Comment:
minor : `hasNonMainBranches()` ?

Reply:
ok, I used a wording used elsewhere but hasNonMainBranches is better